File: LibVt6530/LibVt6530/Cursor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LibVt6530
{
	public class Cursor
	{
		public int m_row, m_column;
		protected int m_numColumns;
		protected int m_numRows;

		public Cursor(int rows, int cols)
		{
			m_row = 0;
			m_column = 0;
			m_numColumns = cols;
			m_numRows = rows;
		}

		public void Clear()
		{
			m_row = 0;
			m_column = 0;
		}

		public void AdjustCol()
		{
			if (m_column >= m_numColumns)
			{
				m_column = 0;
				m_row++;
				AdjustRow();
			}
			if (m_column < 0)
			{
				m_column = 0;
				m_row--;
				AdjustRow();
			}
		}

		public void AdjustRow()
		{
			if (m_row >= m_numRows)
			{
				m_row = 0;
			}
			if (m_row < 0)
			{
				m_row = m_numRows - 1;
			}
		}
	};
}

File: Java/gov/revenue/vt6530/ui/Cursor.java
package gov.revenue.vt6530.ui;

import gov.revenue.ASSERT;

/**
 *  Cursor holds the cursor or buffer position
 *  for a page.  
 */
class Cursor
{
	int row, column;
	int numColumns;
	int numRows;
	
	Cursor(int rows, int cols)
	{
		numColumns = cols;
		numRows = rows;
	}
	
	final void clear()
	{
		row = 0;
		column = 0;
	}

	/**
	 *  Ensure the column is in the display
	 *  area.
	 */
	final void adjustCol()
	{
		if (column >= numColumns)
		{
			column = 0;
			row++;
			adjustRow();
		}
		if (column < 0)
		{
			column = 0;
			row--;
			adjustRow();
		}
	}

	/**
	 *  Ensure the row is in the display area.
	 */
	final void adjustRow()
	{
		if (row >= numRows)
		{
			row = 0;
		}
		if (row < 0)
		{
			row = numRows - 1;
		}
	}
}
